### PR TITLE
Feature: Moved Widget option to settings modal

### LINKED ISSUE #7214: Feature: Move widget options to the main settings modal
### What's the Problem?

Currently it the only settings that is not in the settings menu. I would expected them in there and still end up looking in there for them because it makes more sense. This will add to #7213

** Requirements
- Remove flyout from toolbar
- Add widgets section to the appearance settings page
- Add sub settings to turn widgets on/off

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #7214 

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

File: src/Files.App/ViewModels/SettingsViewModels/AppearanceViewModel.cs [hunk 1]
```diff
@@ -224,7 +224,51 @@ public bool ShowFileTagsSection
             }
         }
 
-        public async Task OpenThemesFolder()
+		#region WidgetsPage Widgets
```
Comment (yaichenbaum) [inline src/Files.App/ViewModels/SettingsViewModels/AppearanceViewModel.cs:227]: We don't use regions anymore

File: src/Files.App/ViewModels/SettingsViewModels/AppearanceViewModel.cs [hunk 2]
```diff
@@ -224,7 +224,51 @@ public bool ShowFileTagsSection
...
+					UserSettingsService.WidgetsSettingsService.ShowBundlesWidget = value;
+			}
+		}
+
+		public bool ShowRecentFilesWidget
+		{
+			get => UserSettingsService.WidgetsSettingsService.ShowRecentFilesWidget;
+			set
+			{
+				if (value != UserSettingsService.WidgetsSettingsService.ShowRecentFilesWidget)
```
Comment (yaichenbaum) [inline src/Files.App/ViewModels/SettingsViewModels/AppearanceViewModel.cs:264]: Settings should be moved to AppearanceSettingsService

Comment (ferrariofilippo) [reply src/Files.App/ViewModels/SettingsViewModels/AppearanceViewModel.cs:264]: Should Widget-Settings be logged?

Comment (yaichenbaum) [reply src/Files.App/ViewModels/SettingsViewModels/AppearanceViewModel.cs]: Yes

File: src/Files.App/ServicesImplementation/Settings/AppearanceSettingsService.cs [hunk 3]
```diff
@@ -117,6 +169,10 @@ public void ReportToAppCenter()
             Analytics.TrackEvent($"{nameof(ShowFileTagsSection)}, {ShowFileTagsSection}");
             Analytics.TrackEvent($"{nameof(PinRecycleBinToSidebar)}, {PinRecycleBinToSidebar}");
             Analytics.TrackEvent($"{nameof(UseCompactStyles)}, {UseCompactStyles}");
-        }
-    }
+			Analytics.TrackEvent($"{nameof(ShowFoldersWidget)}, {ShowFoldersWidget}");
+			Analytics.TrackEvent($"{nameof(ShowRecentFilesWidget)}, {ShowRecentFilesWidget}");
+			Analytics.TrackEvent($"{nameof(ShowDrivesWidget)}, {ShowDrivesWidget}");
+			Analytics.TrackEvent($"{nameof(ShowBundlesWidget)}, {ShowBundlesWidget}");
```
Comment (yaichenbaum) [inline src/Files.App/ServicesImplementation/Settings/AppearanceSettingsService.cs:175]: The `ReportToAppCenter` method is no longer used and was accidentally left in, can you remove it?

File: src/Files.App/ServicesImplementation/Settings/UserSettingsService.cs [hunk 4]
```diff
@@ -108,7 +101,7 @@ public override bool ImportSettings(object import)
         private TSettingsService GetSettingsService<TSettingsService>(ref TSettingsService settingsServiceMember)
             where TSettingsService : class, IBaseSettingsService
         {
-            settingsServiceMember ??= Ioc.Default.GetService<TSettingsService>();
+            settingsServiceMember ??= Ioc.Default.GetService<TSettingsService>()!;
```
Comment (yaichenbaum) [inline src/Files.App/ServicesImplementation/Settings/UserSettingsService.cs:104]: What's this change for?

File: src/Files.Backend/Services/Settings/IAppearanceSettingsService.cs [hunk 5]
```diff
@@ -67,5 +67,33 @@ public interface IAppearanceSettingsService : IBaseSettingsService, INotifyPrope
...
+		/// Gets or sets a value indicating whether or not the drives widget should be visible.
+		/// </summary>
+		bool ShowDrivesWidget { get; set; }
+
+		/// <summary>
+		/// Gets or sets a value indicating whether or not the Bundles widget should be visible.
+		/// </summary>
+		bool ShowBundlesWidget { get; set; }
+
+		bool FoldersWidgetExpanded { get; set; }
```
Comment (yaichenbaum) [inline src/Files.Backend/Services/Settings/IAppearanceSettingsService.cs:91]: Can you add comments for these as well?

File: src/Files.App/Strings/en-US/Resources.resw [hunk 6]
```diff
@@ -1671,33 +1671,24 @@
   <data name="SettingsPreferencesOpenFoldersNewTab.Title" xml:space="preserve">
     <value>Open folders in new tab</value>
   </data>
-  <data name="NavToolbarShowBundlesWidget.Header" xml:space="preserve">
+  <data name="ShowBundlesWidgetSetting.Title" xml:space="preserve">
```
Comment (yaichenbaum) [inline src/Files.App/Strings/en-US/Resources.resw:1674]:
```suggestion
  <data name="ShowBundlesWidget" xml:space="preserve">
```

File: src/Files.App/Strings/en-US/Resources.resw [hunk 7]
```diff
@@ -1671,33 +1671,24 @@
...
     <value>Show Bundles widget</value>
   </data>
-  <data name="NavToolbarShowDrivesWidget.Header" xml:space="preserve">
+  <data name="ShowDrivesWidgetSetting.Title" xml:space="preserve">
```
Comment (yaichenbaum) [inline src/Files.App/Strings/en-US/Resources.resw:1677]:
```suggestion
  <data name="ShowDrivesWidget" xml:space="preserve">
```

File: src/Files.App/Strings/en-US/Resources.resw [hunk 8]
```diff
@@ -1671,33 +1671,24 @@
...
     <value>Show Drives widget</value>
   </data>
-  <data name="NavToolbarShowFolderWidget.Header" xml:space="preserve">
+  <data name="ShowFolderWidgetSetting.Title" xml:space="preserve">
```
Comment (yaichenbaum) [inline src/Files.App/Strings/en-US/Resources.resw:1680]:
```suggestion
  <data name="ShowFolderWidget" xml:space="preserve">
```

File: src/Files.App/Strings/en-US/Resources.resw [hunk 9]
```diff
@@ -1671,33 +1671,24 @@
...
     <value>Show Folders widget</value>
   </data>
-  <data name="NavToolbarShowRecentFilesWidget.Header" xml:space="preserve">
+  <data name="ShowRecentFilesWidgetSetting.Title" xml:space="preserve">
```
Comment (yaichenbaum) [inline src/Files.App/Strings/en-US/Resources.resw:1683]:
```suggestion
  <data name="ShowRecentFilesWidget" xml:space="preserve">
```

File: src/Files.App/Strings/en-US/Resources.resw [hunk 10]
```diff
@@ -1956,7 +1947,7 @@
   <data name="OngoingTasksDismissBannerButton.ToolTipService.ToolTip" xml:space="preserve">
     <value>Dismiss</value>
   </data>
-  <data name="NavToolbarWidgets.Label" xml:space="preserve">
+  <data name="WidgetsSettings.Title" xml:space="preserve">
```
Comment (yaichenbaum) [inline src/Files.App/Strings/en-US/Resources.resw:1950]:
```suggestion
  <data name="Widgets" xml:space="preserve">
```

File: src/Files.App/ServicesImplementation/Settings/UserSettingsService.cs (hunk 4 above)
Comment (ferrariofilippo) [reply src/Files.App/ServicesImplementation/Settings/UserSettingsService.cs:104]: Remove nullable annotation (`Ioc.Default.GetService()` might return null)

File: src/Files.App/Strings/en-US/Resources.resw [hunk 11]
```diff
@@ -1671,33 +1671,24 @@
   <data name="SettingsPreferencesOpenFoldersNewTab.Title" xml:space="preserve">
     <value>Open folders in new tab</value>
   </data>
-  <data name="NavToolbarShowBundlesWidget.Header" xml:space="preserve">
+  <data name="ShowBundlesWidget" xml:space="preserve">
     <value>Show Bundles widget</value>
```
Comment (yaichenbaum) [inline src/Files.App/Strings/en-US/Resources.resw:1675]:
```suggestion
    <value>Bundles</value>
```

File: src/Files.App/Views/SettingsPages/Appearance.xaml [hunk 12]
```diff
@@ -154,6 +154,43 @@
 					IsOn="{Binding UseCompactStyles, Mode=TwoWay}"
 					Style="{StaticResource RightAlignedToggleSwitchStyle}" />
 			</local:SettingsBlockControl>
+
```
Comment (yaichenbaum) [inline src/Files.App/Views/SettingsPages/Appearance.xaml:157]: Can you move the widget settings before the compact styles?

**Review by yaichenbaum — APPROVED** [review]
LGTM

File: src/Files.App/ServicesImplementation/Settings/AppearanceSettingsService.cs [hunk 13]
```diff
@@ -99,24 +146,15 @@ protected override void RaiseOnSettingChangedEvent(object sender, SettingChanged
                 case nameof(ShowFileTagsSection):
                 case nameof(PinRecycleBinToSidebar):
                 case nameof(UseCompactStyles):
+                case nameof(ShowFoldersWidget):
+                case nameof(ShowRecentFilesWidget):
+                case nameof(ShowDrivesWidget):
+                case nameof(ShowBundlesWidget):
                     Analytics.TrackEvent($"{e.SettingName} {e.NewValue}");
```
Comment (yaichenbaum) [inline src/Files.App/ServicesImplementation/Settings/AppearanceSettingsService.cs:153]:
```suggestion
                    Analytics.TrackEvent($"Set {e.SettingName} to {e.NewValue}");
```

File: src/Files.Backend/Services/Settings/IAppearanceSettingsService.cs [hunk 14]
```diff
@@ -67,5 +67,45 @@ public interface IAppearanceSettingsService : IBaseSettingsService, INotifyPrope
...
+		bool RecentFilesWidgetExpanded { get; set; }
+
+		/// <summary>
+		/// Gets or sets a value indicating wheter or not the drives widget section is expanded.
+		/// </summary>
+		bool DrivesWidgetExpanded { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating wheter or not the Bundles widget section is expanded.
+        /// </summary>
```
Comment (yaichenbaum) [inline src/Files.Backend/Services/Settings/IAppearanceSettingsService.cs:108]: Is spacing messed up?

**Review by yaichenbaum — APPROVED** [review]
LGTM
@d2dyno1 can you review this as well?

**Pull request merged**